### DISCUSSION TITLE
Getting started: The request is missing an Authorization header value

### DISCUSSION THREAD

**Discussion by theunreal** [gladly-team/next-firebase-auth#346] [Q&A]
**Describe the bug**
After completing the getting started guide (in this repo), accessing `http://localhost:3000/api/login` shows:

`{"error":"Unexpected error."}`

How I am supposed to work with the login endpoint from my app?

**Versions**

`next-firebase-auth` version: 
next-firebase-auth version: 0.13.5
Firebase JS SDK: 9.5
Next.js: 12

**To Reproduce**

**Expected behavior**

**Additional context**

**Comment by kmjennison** [comment]
The Authorization header is set automatically by default. Have you tried the example app in this repo? Can you share your config? 

If you set a custom `tokenChangedHandler` in your config, you're responsible for passing the Authorization value. Here's a custom `tokenChangedHandler` example: https://github.com/gladly-team/tab-web/blob/master/src/utils/auth/initAuth.js#L22

**Reply by theunreal** [reply]
> The Authorization header is set automatically by default. Have you tried the example app in this repo? Can you share your config?
> 
> If you set a custom `tokenChangedHandler` in your config, you're responsible for passing the Authorization value. Here's a custom `tokenChangedHandler` example: https://github.com/gladly-team/tab-web/blob/master/src/utils/auth/initAuth.js#L22

Hi, thanks for the answer -but it's still not clear to me how it works.
I haven't set any `tokenChangedHandler ` . After setting my initAuth, it's not clear to me how I should connect and make a login flow with a button to create an account on Firebase Auth. What I want to achieve is simple and the getting started guide couldn't help me - login button that saves the user in Firebase Auth (with Google or w/e)

**Reply by kmjennison** [reply]
Please try setting up the example app and share details about where you get stuck. The example app has a [live demo here](https://nfa-example.vercel.app/).

This library doesn't help with the login user interface, like buttons. Use [firebaseui-web](https://github.com/firebase/firebaseui-web) or your own UI for it.

**Reply by DwayneGit** [reply]
Hello, I am getting this issue for the logout. the log out handler is the same as in the example and sign out is through AuthUser.signOut(). everything else seems to work and the user appears to be signed out but i get this error
